File: src/main/java/com/example/service/WorkoutService.java
package com.example.service;

import java.util.List;

import com.example.model.Workout;

public interface WorkoutService {
	
	Workout addWorkout(Workout workout);
	Workout getWorkoutById(Integer workoutId);
	List<Workout> getAllWorkouts();
	Workout updateWorkout(Integer workoutId,Workout workout);
	String deleteWorkout(Integer id);
	//List<Workout> getWorkoutsByUserId(Integer userId);
}

File: Frontend/src/app/services/Workout-service.ts
import { Injectable } from '@angular/core';
import { environment } from './baseUrl';
import { HttpClient } from '@angular/common/http';

@Injectable({
  providedIn: 'root'
})
export class WorkoutService {
  private apiUrl = `${environment.apiBaseUrl}/api/workOuts`;
  constructor(private http :HttpClient) { }

  addWorkout(workout: any) {
      return this.http.post(`${this.apiUrl}`, workout);
  }
  getWorkoutById(workoutId: number) {
    this.http.get(`${this.apiUrl}/getbyid/${workoutId}`);
  }
  getAllWorkouts() {
    return this.http.get(`${this.apiUrl}/getall`);
  }
  updateWorkout(workoutId: number, workout: any) {
    return this.http.put(`${this.apiUrl}/${workoutId}`, workout);
  }
  deleteWorkout(workoutId: number) {
    return this.http.delete(`${this.apiUrl}/${workoutId}`);
  }
}
